File: solutions/fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let (|Between|) low high (x: int) : bool =
    x >= low && x <= high

let successRate (speed: int): float =
    match speed with
    | 0 -> 0
    | Between 1 4 true -> 1.00
    | Between 5 8 true -> 0.90
    | 9 -> 0.80
    | 10 -> 0.77
    | _ -> failwith "Out of range"

let productionRatePerHour (speed: int): float =
    221.0 * (float speed) * successRate speed

let workingItemsPerMinute (speed: int): int =
    productionRatePerHour speed / 60.0 |> int

File: solutions/julia/cars-assemble/cars-assemble.jl
const base_rate = 221

function success_rate(speed)
    speed == 0 ? 0 :
    speed ∈ 1:4 ? 1.00 :
    speed ∈ 5:8 ? 0.90 :
    speed == 9 ? 0.80 :
    0.77
end

function production_rate_per_hour(speed)
    base_rate * speed * success_rate(speed)
end

function working_items_per_minute(speed)
    floor(Int, production_rate_per_hour(speed) / 60)
end
